core/templates/pages/collection-editor-page/services/collection-linearizer.service.ts

```diff
@@ -18,149 +18,185 @@
  * retrieving the collection, saving it, and listening for changes.
  */
 
-require('domain/collection/collection-update.service.ts');
-
-angular.module('oppia').factory('CollectionLinearizerService', [
-  'CollectionUpdateService',
-  function(CollectionUpdateService) {
-    var _getNextExplorationId = function(collection, completedExpIds) {
-      var explorationIds = collection.getExplorationIds();
-
-      for (var i = 0; i < explorationIds.length; i++) {
-        if (completedExpIds.indexOf(explorationIds[i]) === -1) {
-          return explorationIds[i];
-        }
+import { Injectable } from '@angular/core';
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { CollectionNode } from 'domain/collection/collection-node.model';
+import { CollectionUpdateService } from 'domain/collection/collection-update.service';
+import { Collection } from 'domain/collection/collection.model';
+import { LearnerExplorationSummaryBackendDict } from 'domain/summary/learner-exploration-summary.model';
+
+interface SwapFunction {
+  (
+    collection: Collection,
+    linearNodeList: CollectionNode[], nodeIndex: number): void;
+  (
+    collection: Collection,
+    linearNodeList: CollectionNode[], nodeIndex: number): void;
+  (arg0: Collection, arg1: CollectionNode[], arg2: number): void;
+ }
+
+@Injectable({
+  providedIn: 'root'
+})
+export class CollectionLinearizerService {
+  explorationIds: string | string[];
```

Comment:
how can this be both string and string[]? Are you sure about this?

Reply:
fixed